pkg/templates/templates.go

```diff
@@ -142,19 +151,20 @@ func NewResolver(kubeClient *kubernetes.Interface, kubeConfig *rest.Config, conf
 	}, nil
 }
 
-// HasTemplate performs a simple check for the template start delimiter to
-// indicate if the input byte slice has a template. If the startDelim argument is
-// an empty string, the default start delimiter of "{{" will be used.
-func HasTemplate(templateStr []byte, startDelim string) bool {
+// HasTemplate performs a simple check for the template start delimiter or the "$ocm_encrypted" prefix to
+// indicate if the input byte slice has a template. If the startDelim argument is an empty string, the
+// default start delimiter of "{{" will be used.
+func HasTemplate(template []byte, startDelim string) bool {
 	if startDelim == "" {
 		startDelim = defaultStartDelim
 	}
 
+	templateStr := string(template)
 	glog.V(glogDefLvl).Infof("HasTemplate template str:  %v", templateStr)
 	glog.V(glogDefLvl).Infof("Checking for the start delimiter:  %s", startDelim)
 
 	hasTemplate := false
-	if strings.Contains(string(templateStr), startDelim) {
+	if strings.Contains(templateStr, startDelim) || strings.Contains(templateStr, protectedPrefix) {
```

Comment:
In ConfigPolicyController , there is a check for Hub templates , if a policy that came down to managed-cluster still has unresolved hub templates in it then it presumed that an error occurred during HubTemplates processing. 

I believe the either or check at #167 will interfere with this check for hub-template errors.

Reply:
You are right. I added an additional argument to the `HasTemplate` function in the latest push. It's not ideal that it breaks the API, but I prefer this approach over having a second function just for checking for encrypted strings. We'll just need to version this next go-template-utils tag accordingly.